ci/build-test/conda.sh

```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 ./ci/update-versions.sh "${RELEASE_VERSION:-}"
 
-mamba install -y boa conda-build
-conda mambabuild \
+mamba install -y conda-build
+conda build \
```

Comment:
As this was installing `boa` itself, changes in the CI image would not have affected it. Still given `boa` is deprecated, it would be good to clean this up

xref: https://github.com/mamba-org/boa/pull/403